libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -10,9 +16,18 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store) {}
+  constructor(
+    private readonly store: Store,
+    private _snackBar: MatSnackBar
+  ) { }
 
-  removeFromReadingList(item) {
+  removeFromReadingList(item: ReadingListItem) {
+    const mySnackBar = this._snackBar.open(`${item.title} removed from reading list`, 'Undo');
+    mySnackBar.afterDismissed().subscribe(info => {
+      if (info.dismissedByAction === true) {
+        this.store.dispatch(addToReadingList({ book: { id: item.bookId, ...item } }));
+      }
+    });
```

Comment:
You are displaying the Snackbar simultaneously with dispatching the `removeFromReadingList ` action. What happens if the `removeFromReadingList ` action fails? Shouldn't you confirm this action before displaying the snackbar?

Reply:
Just committed a fix, I am subscribing to confirmedRemoveFromReadingList before launching the Snackbar, however I am not sure if this is the best approach.